timemachine/fe/chiral_utils.py

```diff
@@ -132,6 +142,131 @@ def find_chiral_atoms(mol):
     return chiral_atoms
 
 
+def setup_all_chiral_atom_restr_idxs(mol, conf) -> List[FourTuple]:
+    """Apply setup_chiral_atom_restraints to all atoms found by find_chiral_atoms"""
+    chiral_atom_set = find_chiral_atoms(mol)
+    chiral_atom_restr_idxs = []
+    for a_idx in chiral_atom_set:
+        idxs = setup_chiral_atom_restraints(mol, conf, a_idx)
+        for ii in idxs:
+            assert ii not in chiral_atom_restr_idxs
+        chiral_atom_restr_idxs.extend(idxs)
+    return chiral_atom_restr_idxs
+
+
+@dataclass
+class ChiralRestrIdxSet:
+    """Support fast checks of whether a given 4-tuple is disallowed"""
+
+    restr_idxs: List[FourTuple]
+
+    @classmethod
+    def from_mol(cls, mol, conf):
+        restr_idxs = setup_all_chiral_atom_restr_idxs(mol, conf)
+        return ChiralRestrIdxSet(restr_idxs)
+
+    def expand_symmetries(self) -> Tuple[Set[FourTuple], Set[FourTuple]]:
+        allowed_set = set()
+        disallowed_set = set()
+
+        for (center, i, j, k) in self.restr_idxs:
+            # rotations
+            allowed_set.add((center, i, j, k))
+            allowed_set.add((center, j, k, i))
+            allowed_set.add((center, k, i, j))
+
+            # swaps
+            disallowed_set.add((center, i, k, j))
+            disallowed_set.add((center, j, i, k))
+            disallowed_set.add((center, k, j, i))
+
+        assert allowed_set.isdisjoint(disallowed_set)
+
+        return allowed_set, disallowed_set
+
+    def __post_init__(self):
+        self.allowed_set, self.disallowed_set = self.expand_symmetries()
+
+    def defines(self, trial_tuple: FourTuple) -> bool:
+        return (trial_tuple in self.allowed_set) or (trial_tuple in self.disallowed_set)
+
+    def disallows(self, trial_tuple: FourTuple) -> bool:
+        return trial_tuple in self.disallowed_set
+
+
+def _find_atom_map_chiral_conflicts_one_direction(
+    core: np.ndarray,
+    chiral_set_a: ChiralRestrIdxSet,
+    chiral_set_b: ChiralRestrIdxSet,
+    mode="flip",
```

Comment:
nit: You could create constants for the modes (CHIRAL_MODE_FLIP, CHIRAL_MODE_UNDEFINED), since they are used repeatedly.

Reply:
The suggestion to use a named constant instead of a string is good, but this setting seems too specific to add to global constants.py -- added an enum instead https://github.com/proteneer/timemachine/pull/908/commits/3e8738118197b6826f1715a0d3d426b67bda33c8

(and removed an option in `get_core_by_mcs`, to better enforce reasonable default)